### PR TITLE
feat: use `go-ipfs` instead of `js-ipfs`

### PR DESCRIPTION
**Pull request by smrz2001**

### CODE REVIEW

File: modules/ecs/ceramic/security_groups.tf [hunk 1]
```diff
@@ -20,33 +20,11 @@ resource "aws_security_group" "efs" {
...
-  tags = local.default_tags
-}
-
 module "ceramic_security_group" {
   source  = "terraform-aws-modules/security-group/aws"
-  version = "3.0"
+  version = "~> 3.0"
 
-  name        = local.namespace
+  name        = "${local.namespace}-ceramic"
```
Comment (v-stickykeys) [inline modules/ecs/ceramic/security_groups.tf:27]: Why is this needed? I thought the namespace would have ceramic in it already but maybe that's just the way we have the local set 🤔

File: modules/ecs/ceramic/versions.tf [hunk 2]
```diff
@@ -3,15 +3,11 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = "3.36.0"
+      version = "~> 3.63.0"
```
Comment (v-stickykeys) [inline modules/ecs/ceramic/versions.tf:6]: Pinning to specific version seems best

**Review by v-stickykeys — CHANGES_REQUESTED** [review]
Some questions

File: examples/ecs/versions.tf [hunk 3]
```diff
@@ -3,7 +3,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = "3.36.0"
+      version = "~> 3.63.0"
```
[the remark this answers is not in the document]
Comment (smrz2001) [reply examples/ecs/versions.tf:6]: `~>` will pin the major and minor versions but allow the patch version to be greater than `0` (in this case), similar to how `~`  is used in NPM. I mainly changed this to be consistent with how we use versions in our main Terraform code, but I don't have a strong opinion.

https://learn.hashicorp.com/tutorials/terraform/versions?in=terraform/1-0#review-example-configuration

File: modules/ecs/ipfs/locals.tf [hunk 4]
```diff
@@ -7,74 +7,69 @@ locals {
...
   swarm_tcp_port   = 4010
-  swarm_ws_port    = var.use_ssl ? 4012 : 4011
+  # TODO: Choose between 4011 and 4012 based on `use_ssl` once `go-ipfs` supports WSS
+  # swarm_ws_port  = var.use_ssl ? 4012 : 4011
+  swarm_ws_port    = 4011
 
-  announce_address_list = var.use_ssl ? "/dns4/${local.domain_name_external}/tcp/${local.swarm_ws_port}/wss,/dns4/${local.domain_name_external}/tcp/${local.swarm_tcp_port}" : "/dns4/${aws_lb.external.dns_name}/tcp/${local.swarm_ws_port}/ws,/dns4/${aws_lb.external.dns_name}/tcp/${local.swarm_tcp_port}"
-  domain_name_external  = "ipfs-${var.base_namespace}-external.${var.domain}"
-  domain_name_internal  = "ipfs-${var.base_namespace}-internal.${var.domain}"
+  announce_address_list = var.use_ssl ? "/dns4/${local.domain_name_external}/tcp/${local.swarm_ws_port}/ws,/dns4/${local.domain_name_external}/tcp/${local.swarm_tcp_port}" : "/dns4/${aws_lb.external.dns_name}/tcp/${local.swarm_ws_port}/ws,/dns4/${aws_lb.external.dns_name}/tcp/${local.swarm_tcp_port}"
```
[the remark this answers is not in the document]
Comment (smrz2001) [reply modules/ecs/ipfs/locals.tf:14]: Yeah, good idea, will remove it for now. I don't see support being available very soon.

File: examples/ecs/main.tf [hunk 5]
```diff
@@ -15,13 +15,14 @@ module "ceramic_ecs" {
 
   count = var.deployments
 
-  acm_certificate_arn            = data.aws_acm_certificate._3boxlabs_com.arn
+  acm_certificate_arn            = var.acm_certificate_arn
```
[the remark this answers is not in the document]
Comment (smrz2001) [reply examples/ecs/main.tf:18]: I was trying to keep the code in sync with our main TF repo. I can currently run `terraform apply` from within the `app-elp` folder in the main repo, or from within the `examples/ecs` folder in this repo using the same `tfvars`.

To keep that consistency, I'll add back the code I removed to both places, the main repo and this one.

File: examples/ecs/versions.tf (hunk 3 above)
[the remark this answers is not in the document]
Comment (smrz2001) [reply examples/ecs/versions.tf:6]: I'll revert my change and pin.

File: modules/ecs/ceramic/security_groups.tf [hunk 6]
```diff
@@ -20,33 +20,11 @@ resource "aws_security_group" "efs" {
...
-    }
-  ]
-
-  tags = local.default_tags
-}
-
 module "ceramic_security_group" {
   source  = "terraform-aws-modules/security-group/aws"
-  version = "3.0"
+  version = "~> 3.0"
```
[the remark this answers is not in the document]
Comment (smrz2001) [reply modules/ecs/ceramic/security_groups.tf:25]: I'll revert my change and pin.

File: modules/ecs/ceramic/versions.tf (hunk 2 above)
Comment (smrz2001) [reply modules/ecs/ceramic/versions.tf:6]: Will revert and pin.

File: modules/ecs/ipfs/versions.tf [hunk 7]
```diff
@@ -3,15 +3,11 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = "3.36.0"
+      version = "~> 3.63.0"
```
[the remark this answers is not in the document]
Comment (smrz2001) [reply modules/ecs/ipfs/versions.tf:6]: Will revert and pin.

File: modules/ecs/versions.tf [hunk 8]
```diff
@@ -3,15 +3,11 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = "3.36.0"
+      version = "~> 3.63.0"
```
[the remark this answers is not in the document]
Comment (smrz2001) [reply modules/ecs/versions.tf:6]: Will revert and pin.

File: examples/ecs/main.tf (hunk 5 above)
[the remark this answers is not in the document]
Comment (smrz2001) [reply examples/ecs/main.tf:18]: Actually, I'll make them optionally configurable.

[the remark this answers is not in the document]
Comment (smrz2001) [reply examples/ecs/main.tf:18]: In the future, I think we should make more/all of the other values also configurable, or even consider creating them within this Terraform instead of needing them to be created out-of-band.

File: modules/ecs/ipfs/locals.tf (hunk 4 above)
[the remark this answers is not in the document]
Comment (smrz2001) [reply modules/ecs/ipfs/locals.tf:14]: Although, that flag is used for more than just swarm stuff, so can't remove it entirely, just not use it for swarm.

File: modules/ecs/ipfs/locals.tf [hunk 9]
```diff
@@ -7,74 +7,69 @@ locals {
...
     local.api_lb_external,
     local.api_lb_internal,
     local.gateway_lb_external,
     local.gateway_lb_internal,
-    local.swarm_lb_external,
-    local.swarm_lb_internal
+    local.swarm_lb_external
   )
 
+  log_level = "info"
```
[the remark this answers is not in the document]
Comment (smrz2001) [reply modules/ecs/ipfs/locals.tf:71]: It gets injected into the `go-ipfs` env and is used as the default log level for IPFS. I'll make it configurable.

File: modules/ecs/ipfs/variables.tf [hunk 10]
```diff
@@ -141,26 +141,11 @@ variable "enable_internal_gateway" {
...
-  type        = bool
-  description = "True to enable IPFS Swarm on internal ALB. Note: Swarm is always enabled for the external ALB."
-}
-
 variable "enable_pubsub" {
   type        = bool
   description = "True to enable IPFS PubSub"
 }
 
-variable "dht_server_mode" {
```
[the remark this answers is not in the document]
Comment (smrz2001) [reply modules/ecs/ipfs/variables.tf:154]: This variable wasn't being used anywhere, not here, and not in the main repo...

[the remark this answers is not in the document]
Comment (smrz2001) [reply modules/ecs/ipfs/variables.tf:154]: Do we know if the DHT is actually enabled lol?

File: modules/ecs/ipfs/locals.tf (hunk 4 above)
[the remark this answers is not in the document]
Comment (v-stickykeys) [reply modules/ecs/ipfs/locals.tf:12]: Ahh right. ok cool

**Comment by v-stickykeys** [comment]
> I was trying to keep the code in sync with our main TF repo. I can currently run terraform apply from within the app-elp folder in the main repo, or from within the examples/ecs folder in this repo using the same tfvars.
@smrz2001 I don't think this should be the goal. In our 3box labs terraform repo we should be able to use tfvars file in whatever way is easiest for us, but in the examples here we should make it as clear as possible to people how to use the module, ideally with as little effort as possible.

File: modules/ecs/ipfs/variables.tf (hunk 10 above)
[the remark this answers is not in the document]
Comment (smrz2001) [reply modules/ecs/ipfs/variables.tf:154]: I see `ipfs stats dht` working and showing things, but not able to find where/how DHT gets enabled (even by default).

[the remark this answers is not in the document]
Comment (smrz2001) [reply modules/ecs/ipfs/variables.tf:154]: Ok, found it (in `go-ipfs/cmd/ipfs/daemon.go`)
```
		cmds.StringOption(routingOptionKwd, "Overrides the routing option").WithDefault(routingOptionDefaultKwd),
```
```
	routingOption, _ := req.Options[routingOptionKwd].(string)
	if routingOption == routingOptionDefaultKwd {
		routingOption = cfg.Routing.Type
		if routingOption == "" {
			routingOption = routingOptionDHTKwd
		}
	}
```

[the remark this answers is not in the document]
Comment (v-stickykeys) [reply modules/ecs/ipfs/variables.tf:154]: Yup enabled by default. all good here

File: modules/ecs/locals.tf [hunk 11]
```diff
@@ -1,3 +1,3 @@
 locals {
-  namespace = "ceramic-${var.env}-${var.base_namespace}"
+  namespace = "ceramic-elp-${var.base_namespace}"
```
Comment (v-stickykeys) [inline modules/ecs/locals.tf:2]: No more ELP 🙌 (now that we are on v2)
```suggestion
  namespace = "ceramic-${var.env}-${var.base_namespace}"
```

**Review by v-stickykeys — APPROVED** [review]
Just minor things but should be good after that

**Comment by smrz2001** [comment]
> > I was trying to keep the code in sync with our main TF repo. I can currently run terraform apply from within the app-elp folder in the main repo, or from within the examples/ecs folder in this repo using the same tfvars.
> 
> @smrz2001 I don't think this should be the goal. In our 3box labs terraform repo we should be able to use tfvars file in whatever way is easiest for us, but in the examples here we should make it as clear as possible to people how to use the module, ideally with as little effort as possible on their part.

Makes sense, @v-stickykeys.

We should work towards adding more resources here as well (e.g. S3 bucket, etc.) that must be created outside of these scripts.

**Comment by v-stickykeys** [comment]
@smrz2001 can you update readme to include instructions about peer id and private key?

File: modules/ecs/locals.tf (hunk 11 above)
Comment (smrz2001) [reply modules/ecs/locals.tf:2]: Lol, yes. We'll have to finagle this to get it to work for the migration though. The `ceramic_env` was set to `elp` so it matched the infra we have now, but using `env` will map everything to `ceramic-prod*` and not `ceramic-elp*`.

Comment (smrz2001) [reply modules/ecs/locals.tf:2]: I can still pass it in as config, just not call it `ceramic_env`.

File: modules/ecs/ipfs/main.tf [hunk 12]
```diff
@@ -49,29 +55,29 @@ resource "aws_ecs_task_definition" "main" {
...
 
       api_port              = local.api_port
       gateway_port          = local.gateway_port
       healthcheck_port      = local.healthcheck_port
       swarm_tcp_port        = local.swarm_tcp_port
       swarm_ws_port         = local.swarm_ws_port
       announce_address_list = local.announce_address_list
-      dht_server_mode       = var.dht_server_mode
-      debug                 = var.debug
+      log_level             = local.log_level
```
[the remark this answers is not in the document]
Comment (smrz2001) [reply modules/ecs/ipfs/main.tf]: Yeah, it just needs to be set in the `IPFS_LOGGING` environment variable:
```
    IPFS_LOGGING - sets the level of verbosity of the logging.
        One of: debug, info, warn, error, dpanic, panic, fatal
```

File: modules/ecs/locals.tf (hunk 11 above)
Comment (v-stickykeys) [reply modules/ecs/locals.tf:2]: ohhhh ok ok. i remember now why I named it that. I guess ceramic_env is better actually
